core/tests/protractor_utils/ExplorationPlayerPage.js

```diff
@@ -320,7 +320,10 @@ var ExplorationPlayerPage = function() {
   // corresponding interaction's protractor utilities.
   // Its definition and type are interaction-specific.
   this.submitAnswer = async function(interactionId, answerData) {
-    await waitFor.visibilityOf(
+    // Skip-linter debugger;.
+    // Skip-linter console.log(await element(By.className(
+    //   'protractor-test-conversation-input')).isDisplayed());.
+    await waitFor.presenceOf(
```

Comment:
@U8NWXD @kevintab95 for some reason conversation input is shown as visible even when the size of `.protractor-test-conversation-input` is 0x0. When I migrated the continue interaction is started resolving to false. I checked and everything is the same (at least it looked the same to me). I will detail out what I tested. I put a debugger statement (as you can see in the highlighted code). Once I code execution paused at this point, I open the chrome dev tools in the browser window and search for `protractor-test-conversation-input` and found it. In both develop and my branch it is 0x0. Then I added the `console.log(await element(By.className(
    //   'protractor-test-conversation-input')).isDisplayed())` right above the waitFor.visiblity statement and it came as true for develop and false for me. Am I missing something?
Test that can be used to check this hypothesis: 
test: `coreEditorAndPlayerFeatures`,
describe block: `Core exploration functionality`
it block: `should display plain text content`

Reply:
I'm not sure I fully understand, but maybe this will help:

* It looks like `isDisplayed()` and `visibilityOf()` don't check for quite the same thing. Here's a quote from the [protractor docs](https://www.protractortest.org/#/api?view=ProtractorExpectedConditions.prototype.visibilityOf):
  > Visibility means that the element is not only displayed but also has a height and width that is greater than 0.
  I think this means that an element can have `isDisplayed()` true even when it has size 0x0.

I'm confused by the fact that the highlighted code uses `presenceOf` instead of `visibilityOf`. Was the investigation you describe done before changing `visibilityOf` to `presenceOf`?

Reply:
@U8NWXD 
> It looks like isDisplayed() and visibilityOf() don't check for quite the same thing. Here's a quote from the protractor docs:

isDisplayed returned for me false even when the size was 0x0.

Let me rephrase what I said before: My problem is when the element is in the exact same position with or without my changes to it (at least that's how it looked to me), how is it that for one case I am getting no errors and for the other I get timeout error for visibilityOf?

> I'm confused by the fact that the highlighted code uses presenceOf instead of visibilityOf. Was the investigation you describe done before changing visibilityOf to presenceOf?

Using `presenceOf` made all the tests pass because `presenceOf` returns true/ that it is present. If you look at the code before the changes, it was `await visibiltyOf`. The reason I am asking this is because I don't want to make changes to tests to make them pass. While presenceOf doesn't sound wrong, I want to be a 100% sure if this is ok and why is migrating to angular causing issues with protractor tests and what can we do to break them less (this happens when the changes are perfectly fine). It happens with almost 50% of the migration PRs that I make and want to get to the root cause of these problems.

Reply:
Hi @srijanreddy98, this is an interesting find and thanks for bringing it to our attention! I have verified this and can confirm that the webdriver method `isDisplayed()` does not work as expected on develop in this particular case / test (this method is what `visibilityOf()` calls internally in addition to `webdriverElement.isPresent()`. see [permalink](https://github.com/angular/protractor/blob/e77885d7a60fd5a56bf624a5307fd41582f10bca/lib/expectedConditions.ts#L393-L396)) i.e. it reports true when it should be false because the element is not visible.

I think it might be a bug. Please take a look at https://github.com/angular/protractor/issues/759#issuecomment-57693485. Maybe if you can reduce this to a reproducible form, you can create an issue on the protractor tracker.

I tried this to confim what Srijan mentioned.

on develop
```
await waitFor.visibilityOf(
  conversationInput, 'Conversation input takes too long to appear.');
console.error(await conversationInput.isPresent());
console.error(await conversationInput.isDisplayed());
debugger;    
```

Both `isPresent` and `isDisplayed` (`visibilityOf` uses both internally) returned true.

on the PR branch
```
try{
  await waitFor.visibilityOf(
     conversationInput, 'Conversation input takes too long to appear.');
}  catch (e) {
  console.error(await conversationInput.isPresent());
  console.error(await conversationInput.isDisplayed());
  debugger;
}    
```

In this scenario, the `visibilityOf` check failed. `isPresent` returned true and `isDisplayed` returned false.

Reply:
It seems like the `conversationInput` element is invisible when the interaction is a Continue button and is visible at other times (need to verify). If so, would it suffice to move the `conversationInput` visibilityOf check into the interaction specific submitAnswer method e.g. in the submitAnswer method for FractionInput interactions?

https://github.com/oppia/oppia/blob/de59e50c59a84369fd04c26c080b15648e775059/extensions/interactions/FractionInput/protractor.js#L33-L36

@U8NWXD WDYT?

Reply:
This sounds good to me!

Reply:
@U8NWXD @kevintab95 What I've done for now is put an if case. if `interactionId === 'continue'` use presenseOf, else use `visibilityOf`. Does that sound ok for a temporary change?